### PR TITLE
S924-008 Fix unwrapping of enum types

### PR DESCRIPTION
**Pull request by danielmercier**
An EnumType has a ctype view, thus, to unwrap it, it is unecessary to
call the unwrap function.

The wrap function was not being affected by the issue.

wrap and unwrap functions are required only if a type does not define a
ctype view. Thus, refactor the code to reflect this.

### CODE REVIEW

File: langkit/ocaml_api.py [hunk 1]
```diff
@@ -253,6 +253,35 @@ def wrap_requires_context(self, type):
             ' (wrapping): {}'.format(type)
         ))
 
+    def have_ctype_view(self, type):
```
Comment (pmderodat) [inline langkit/ocaml_api.py]: I suggest to rename to `has_ctype_view`: we check for a single type, so singular would make more sense

File: langkit/ocaml_api.py [hunk 2]
```diff
@@ -253,6 +253,35 @@ def wrap_requires_context(self, type):
...
+        """
+        Given a type, return true if it has a ctype view. Having a ctype view
+        means that wrap and unwrap operations are not needed explicitely.
+
+        :param ct.CompiledType type: Type we want to check for having a view
+            or not.
```
Comment (pmderodat) [inline langkit/ocaml_api.py]: “The type to check.”?

**Comment by pmderodat** [comment]
Hello Daniel,

Thank you very much for this fix! Do you think you could enhance the `ocaml_api/general` testcase to cover the case your patch fixes?

File: langkit/ocaml_api.py (hunk 1 above)
Comment (danielmercier) [reply langkit/ocaml_api.py]: Sure.. Done

File: langkit/ocaml_api.py (hunk 2 above)
Comment (danielmercier) [reply langkit/ocaml_api.py]: Maybe my sentence was a bit to complex, agreed :p

**Comment by danielmercier** [comment]
Thank for the review, I made the changes and added the test to ocaml_api/general

**Pull request merged**